lib/parse.js

```diff
@@ -144,9 +165,12 @@ module.exports = function (str, opts) {
 
     options.ignoreQueryPrefix = options.ignoreQueryPrefix === true;
     options.delimiter = typeof options.delimiter === 'string' || utils.isRegExp(options.delimiter) ? options.delimiter : defaults.delimiter;
+    options.arrayDelimiter = typeof options.arrayDelimiter === 'string' || utils.isRegExp(options.arrayDelimiter)
+        ? options.arrayDelimiter : defaults.arrayDelimiter;
     options.depth = typeof options.depth === 'number' ? options.depth : defaults.depth;
     options.arrayLimit = typeof options.arrayLimit === 'number' ? options.arrayLimit : defaults.arrayLimit;
     options.parseArrays = options.parseArrays !== false;
+    options.splitArrays = options.splitArrays !== false;
```

Comment:
in this case, `parseArrays` isn't the pattern to follow; this should ensure it's `=== true` instead. As is, this would be a breaking change.

Reply:
Hey thanks for taking the time to respond, I'm going to read through #219 and come back - I'm not dead set on any particular pattern, just happens to be my first pass, so please let me know if you see any potential issues or areas that desire more testing.

Thanks!

Reply:
ah, so a function to split the value into arrays would remove the need for settings sprawl?

Reply:
yes, exactly